x-pack/plugins/apm/server/index.ts

```diff
@@ -69,6 +69,64 @@ const configSchema = schema.object({
     schema.string({ defaultValue: '' }),
     schema.never()
   ),
+  featureFlags: schema.object({
+    agentConfigurationAvailable: schema.conditional(
+      schema.contextRef('serverless'),
+      true,
+      schema.boolean({
+        defaultValue: false,
+      }),
+      schema.oneOf([schema.literal(true)], { defaultValue: true })
+    ),
+    configurableIndicesAvailable: schema.conditional(
+      schema.contextRef('serverless'),
+      true,
+      schema.boolean({
+        defaultValue: false,
+      }),
+      schema.oneOf([schema.literal(true)], { defaultValue: true })
+    ),
```

Comment:
nit: for simplicity (especially on reading), may I suggest we define the validator in a variable and reuse it for all the feature flags?

```ts
const disabledOnServerless = schema.conditional(
  schema.contextRef('serverless'),
  true,
  schema.boolean({
    defaultValue: false,
  }),
  schema.oneOf([schema.literal(true)], { defaultValue: true })
);

const configSchema = schema.object({
  ...
  featureFlags: schema.object({
    agentConfigurationAvailable: disabledOnServerless,
    configurableIndicesAvailable: disabledOnServerless,
    infrastructureTabAvailable: disabledOnServerless,
    ...
  })
});
```

WDYT?

Reply:
Thanks for the suggestion @afharo, this looks a lot better